src/ast/passes/codegen_llvm.cpp

```diff
@@ -1881,16 +1881,17 @@ void CodegenLLVM::visit(Ternary &ternary)
                                                "right",
                                                parent);
   BasicBlock *done = BasicBlock::Create(module_->getContext(), "done", parent);
+
+  Value *buf = nullptr;
+  if (ternary.type.IsStringTy()) {
```

Comment:
Maybe I'm missing something but why can't this logic be nested inside the `ternary.type.IsStringTy()` on line 1921?

Reply:
Yeah, I tried that initially but LLVM barfs during IR validation. It's b/c there's a CreateCondBr on line 1896, and if we put this initialization code before the insert points for left_block and right_block, we strand code between basic blocks. I wanted to keep the current code structure so I put the stuff up here

Reply:
Ah ok. Maybe just move this new code under the "// ordering of all the following statements is important" comment - as it seems to be important for this too

Reply:
Yeah good call